### PR TITLE
Pacman plugin: allow bypass ignored packages with force param

### PR DESCRIPTION
**Pull request by Rumb0**
##### SUMMARY

Add new feature for pacman plugin `force` option.
When set to `true`: allow install/upgrade for packages specified in pacman.conf `IgnorePkg` option 

Issue: [5006](https://github.com/ansible-collections/community.general/issues/5006)

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

pacman

### CODE REVIEW

File: changelogs/fragments/5006-pacman-force-bypass-ignorepkgs.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - pacman plugin - added bypass for IgnorePkg pacman option with ``force`` flag
```
Comment (felixfontein) [inline changelogs/fragments/5006-pacman-force-bypass-ignorepkgs.yml:2]:
```suggestion
  - pacman - added bypass for ``IgnorePkg`` pacman option with ``force`` flag (https://github.com/ansible-collections/community.general/pull/5805).
```

**Review by felixfontein — COMMENTED** [review]
Isn't this change changing the behavior of the module, and thus making a breaking change? If it is, it needs to be controlled by a new option, since we do not merge breaking changes without very good reason (and usually a long deprecation cycle).

**Comment by felixfontein** [comment]
CC @jraby

**Comment by jraby** [comment]
Thanks for looking into this @Rumb0 .

I think @felixfontein is right, will change behavior for playbooks that used both upgrade and force. (or present and force, or latest and force)

It depends what's ansible policy on changing undocumented behavior I guess.

I don't mind either way, going with `force` as is (since `force` was never documented to work with `upgrade` or `present` or `latest`)  or using another bool (`ignore_ignored_packages`  just kidding)

I think it is unfortunate that I missed that use case, using `force: true` in this manner makes sense.

In any case the code reads ok.  I would love a test for it, if that's at all possible.

**Comment by felixfontein** [comment]
> It depends what's ansible policy on changing undocumented behavior I guess.

This depends a lot on for how long that behavior has been in place, and how wrong it seems. If it has been in place for a long time, or is basically a valid behavior (though maybe not something folks would expect or hope for), it must not be changed without a deprecation period (which includes a deprecation warning).

Since the current behavior isn't wrong (in the sense that we should consider it like a bug that needs to be fixed), I would not change it. Instead I would:
1. document what force does in other situations (should not be done in this PR so it can be backported to older stable branches), and
2. add a new option to control this behavior (in this PR).

**Comment by felixfontein** [comment]
ping @Rumb0 

needs_info

**Comment by Rumb0** [comment]
I agree with adding a new module parameter. Something like that would suffice:

```yaml
    ignorepkg_policy:
        description:
            - This controls behavior towards packages listed in IgnorePkg pacman configuration option.
            - C(ignore) - will not upgrade ignored packages on specific package operation nor system upgrade.
            - "C(inherit) - will inherit pacman client behavior: ignore listed packages on system upgrade but upgrade them on specific package operation."
            - If not specified, it defaults to C(ignore).
        default: 'ignore'
        choices: [ ignore, inherit ]
        type: str
        version_added: ?????

```

**Review by russoz — COMMENTED** [review]
Hi there, a small nit to pick in the docs.

**Comment by felixfontein** [comment]
needs_info

**Pull request closed without merging**